font/src/ft/mod.rs

```diff
@@ -206,6 +206,18 @@ struct FullMetrics {
     cell_width: f64,
 }
 
+#[derive(Hash, Eq, PartialEq, Debug)]
+struct FTFaceLocation {
+    path: PathBuf,
+    index: isize,
```

Comment:
What exactly is this mysterious `index` and why is it necessary to keep that around as part of the font path?

Is this related to OTF fonts and it's basically a selection for which face inside a path should be loaded?

Also couldn't this entire `FTFaceLocation` struct be moved to `pattern.rs`, so we would only have a `.location()` method instead of having to load both the path and the index, introducing the need to combine them whenever calling at and adding potential for misuse?

Reply:
>What exactly is this mysterious index and why is it necessary to keep that around as part of the font path?

It's an index of a face in a font file. Most of the time it's just 0, but for variable fonts it could have multiple of them. So index X will be for normal face, but index Y for bold face.

>Is this related to OTF fonts and it's basically a selection for which face inside a path should be loaded?

Yes.

>Also couldn't this entire FTFaceLocation struct be moved to pattern.rs, so we would only have a .location() method instead of having to load both the path and the index, introducing the need to combine them whenever calling at and adding potential for misuse?

Probably, it could simplify logic a bit, since we need both of them.

Reply:
Sure, let's move it then to make it a little easier to use.